src/app/business/block/bucket-detail/bucket-detail.component.ts

```diff
@@ -101,6 +105,76 @@ export class BucketDetailComponent implements OnInit {
       this.getTypes();
     }
     );
+    this.copySelectedDir = window.sessionStorage['searchIndex'] != "" ? JSON.parse(window.sessionStorage.getItem("searchIndex")) : [];
+  }
+  clickOperate(){
+    if(this.selectedDir.length >0){
+      this.isReadyCopy = false;
+      this.copySelectedDir = lodash.cloneDeep(this.selectedDir);
+    }else{
+      this.isReadyCopy = true;
+      this.copySelectedDir = window.sessionStorage['searchIndex'] != "" ? JSON.parse(window.sessionStorage.getItem("searchIndex")) : [];
+      this.isReadyPast = this.copySelectedDir.length != 0 ? false :true;
+    }
+    this.moreItems = [
+      {
+        disabled: this.isReadyCopy,
+        label: 'copy', command: () => {
+          this.copyObject();
+        }
+      },
+      {
+        disabled: this.isReadyPast,
+        label: 'paste', command: (y) => {
+          this.pasteObject();
+        }
+      },
+    ];
+  }
+  //copy object
+  copyObject() {
+    this.isReadyPast = true;
+    this.copySelectedDir.forEach((item, index) =>{
+      this.copySelectedDir[index]['source'] = this.bucketId;
+      this.copySelectedDir[index]['folderId'] = this.folderId;
+    })
+    window.sessionStorage['searchIndex'] = JSON.stringify(this.copySelectedDir);
+  }
+  //paste object
+  pasteObject() {
+    this.copySelectedDir.forEach(item=>{
+      let key = this.folderId != "" ? this.folderId + item.Key : item.Key;
+      let copySource = item.folderId != "" ? item.source + '/' + item.folderId + item.Key : 
+      item.source + '/' + item.Key;
+      let sourceBucket = item.source;
+      window['getAkSkList'](() => {
+        let requestMethod = "PUT";
+        let url = this.BucketService.url + "/" + this.bucketId + '/' + key;
+        window['canonicalString'](requestMethod, url, () => {
+          let options: any = {};
+          this.getSignature(options);
+          options.headers.set('Content-Type', 'application/xml');
+          //The source data
+          options.headers.set('x-amz-copy-source', copySource);
+          if(sourceBucket != this.bucketId){
+            //Copy between different buckets
+            options.headers.set('X-Amz-Metadata-Directive', 'COPY');
```

Comment:
Is this AWS specific header? Or is this header required for backend for copy/paste ?

Reply:
This header parameter is required to distinguish between different buckets or within the same bucket